client/homebrew/pages/editPage/editPage.jsx

```diff
@@ -52,35 +57,81 @@ const EditPage = createClass({
 
 	getInitialState : function() {
 		return {
-			brew                   : this.props.brew,
+			brew : {
+				text        : this.props.brew.text || '',
+				style       : this.props.brew.style || '',
+				shareId     : this.props.brew.shareId || null,
+				editId      : this.props.brew.editId || null,
+				gDrive      : this.props.brew.gDrive || false,
+				title       : this.props.brew.title || '',
+				description : this.props.brew.description || '',
+				tags        : this.props.brew.tags || '',
+				published   : this.props.brew.published || false,
+				authors     : this.props.brew.authors || [],
+				systems     : this.props.brew.systems || [],
+				renderer    : this.props.brew.renderer || 'legacy'
+			},
```

Comment:
```suggestion
			brew : this.props.brew,			
```
Leave this as it was. Both `/edit/:id` and `new/:id` should come with a `this.props.brew` object that already has all of these filled. `new/` doesn't get any brew object from the server, so `this.props.brew` instead uses the `getDefaultProps()` value above. In either case, by the time we get to this step, we have a fully valid `this.props.brew` and we don't need to copy each individual brew property component line by line anymore.

I know several of our pages have something like this where the props are copied line by line back into `this.state`, but I think that was a misunderstanding of Scott when he first built this years ago.